mlir/lib/Analysis/DataFlow/IntegerRangeAnalysis.cpp

```diff
@@ -37,6 +37,24 @@
 using namespace mlir;
 using namespace mlir::dataflow;
 
+namespace mlir::dataflow {
+LogicalResult staticallyNonNegative(DataFlowSolver &solver, Value v) {
```

Comment:
I'd make this return a plain `bool` -- not sure if we gain anything from `LogicalResult`.

Reply:
it needs to be `std::optional<bool>` but same thing in my book - I'll make the change.

Reply:
ah, then keep as-is